src/graphql/pull_request.rs

```diff
@@ -3,36 +3,232 @@ use std::fmt;
+pub(crate) struct Review {
+    pub(crate) author: Option<User>,
+    pub(crate) state: PullRequestReviewState,
+    pub(crate) body: Option<String>,
+    pub(crate) url: String,
+    #[graphql(name = "createdAt")]
+    pub(crate) created_at: DateTimeUtc,
+    #[graphql(name = "publishedAt")]
+    pub(crate) published_at: DateTimeUtc,
+    #[graphql(name = "submittedAt")]
+    pub(crate) submitted_at: DateTimeUtc,
+    #[graphql(name = "isMinimized")]
+    pub(crate) is_minimized: bool,
+    pub(crate) comments: Vec<Comment>,
+}
+
+#[derive(SimpleObject)]
+pub(crate) struct CommitInfo {
+    pub(crate) additions: i32,
+    pub(crate) deletions: i32,
+    pub(crate) message: String,
+    #[graphql(name = "messageBody")]
+    pub(crate) message_body: Option<String>,
+    pub(crate) author: Option<User>,
+    #[graphql(name = "changedFilesIfAvailable")]
+    pub(crate) changed_files_if_available: Option<i32>,
+    #[graphql(name = "committedDate")]
+    pub(crate) committed_date: DateTimeUtc,
+    pub(crate) committer: Option<User>,
+}
 
 #[derive(SimpleObject)]
 pub(crate) struct PullRequest {
-    pub(crate) owner: String,
-    pub(crate) repo: String,
+    pub(crate) id: ID,
     pub(crate) number: i32,
     pub(crate) title: String,
-    pub(crate) assignees: Vec<String>,
-    pub(crate) reviewers: Vec<String>,
+    pub(crate) body: Option<String>,
+    pub(crate) state: PullRequestState,
+    #[graphql(name = "assignees")]
+    pub(crate) assignees: Vec<User>,
+    #[graphql(name = "createdAt")]
+    pub(crate) created_at: DateTimeUtc,
+    #[graphql(name = "updatedAt")]
+    pub(crate) updated_at: DateTimeUtc,
+    #[graphql(name = "closedAt")]
+    pub(crate) closed_at: Option<DateTimeUtc>,
+    #[graphql(name = "mergedAt")]
+    pub(crate) merged_at: Option<DateTimeUtc>,
+    pub(crate) author: Option<User>,
+    pub(crate) additions: i32,
+    pub(crate) deletions: i32,
+    pub(crate) url: String,
+    pub(crate) repository_owner: User,
+    pub(crate) repository_name: String,
+    pub(crate) labels: Vec<Label>,
+    #[graphql(name = "commentsCount")]
+    pub(crate) comments_count: i32,
+    pub(crate) comments: Vec<Comment>,
+    #[graphql(name = "reviewDecision")]
+    pub(crate) review_decision: Option<PullRequestReviewState>,
+    #[graphql(name = "reviewRequests")]
+    pub(crate) review_requests: Vec<User>,
+    #[graphql(name = "reviewsCount")]
+    pub(crate) reviews_count: i32,
+    pub(crate) reviews: Vec<Review>,
+    #[graphql(name = "commitsCount")]
+    pub(crate) commits_count: i32,
+    pub(crate) commits: Vec<CommitInfo>,
+    // retained for key parsing
+    #[graphql(skip)]
+    owner: String,
+    #[graphql(skip)]
+    repo: String,
 }
 
 impl TryFromKeyValue for PullRequest {
+    #[allow(clippy::too_many_lines)]
     fn try_from_key_value(key: &[u8], value: &[u8]) -> anyhow::Result<Self> {
-        let (owner, repo, number) = database::parse_key(key)
+        let (owner, repo, _number) = database::parse_key(key)
```

Comment:
Since we are using `database::parse_key` function in graphql::pull_request, we should change integer return type of `database::parse_key`.